site/templates/_assertions-menu.ejs

```diff
@@ -1,6 +1,16 @@
 <nav id="assertion-menu" class="sidebar js-remember-scroll-position">
     <ul>
-        <% Object.keys(assertionsByType).sort().forEach(function (type) { %>
+        <%
+        var order = { any: 0, boolean: 1, number: 2, string: 3, array: 4, function: 5, object: 6, Error: 7 };
+        Object.keys(assertionsByType).sort(function (a, b) {
+            if (typeof order[a] === 'number' && (typeof order[b] === 'undefined' || order[a] < order[b])) {
+                return -1;
+            } else if (typeof order[b] === 'number' && (typeof order[a] === 'undefined' || order[a] > order[b])){
+                return 1;
+            } else {
+                return 0;
+            }
+        }).forEach(function (type) { %>
```

Comment:
We should make this sort in build.js instead. See: https://github.com/unexpectedjs/unexpected/blob/master/site/build.js#L60-L66